script.js

```diff
@@ -0,0 +1,185 @@
+ * The template is shown below:
+ *   <smart-dialog class="modal-dialog">
+ *     <div class="outer-container">
+ *       <div class="container">
+ *         <div class="modal-dialog-message">
+ *         </div>
+ *         <div class="modal-dialog-button-group">
+ *           <smart-button></smart-button>
+ *           ...
+ *           <smart-button></smart-button>
+ *         </div>
+ *       </div>
+ *     </div>
+ *   </smart-dialog>
+ */
+
+(function(exports) {
+  'use strict';
+
+  function SmartModalDialog() {
+
+    // Determine whether this dialog is opened or not
+    this.isOpened = false;
+    this._clickedIndex = null;
+
+    this.smartBubble = document.createElement('smart-bubbles');
+    document.body.appendChild(this.smartBubble);
+
+    this.element = document.createElement('smart-dialog');
+    this.element.classList.add('modal-dialog');
+    // make dialog focusable in order to catch focus from mouse click or touch
+    // event.
+    this.element.setAttribute('tabIndex', '-1');
+
+    // in order to make vertical align, we need additional container
+    this.outerContainer = document.createElement('div');
+    this.outerContainer.classList.add('outer-container');
+
+    this.container = document.createElement('div');
+    this.container.classList.add('container');
+
+    this.messageElement = document.createElement('div');
+    this.messageElement.classList.add('modal-dialog-message');
+
+    this.buttonGroup = document.createElement('div');
+    this.buttonGroup.classList.add('modal-dialog-button-group');
+    this.buttonGroup.setAttribute('smart-bubbles', 'true');
+
+    this.container.appendChild(this.messageElement);
+    this.container.appendChild(this.buttonGroup);
+
+    this.outerContainer.appendChild(this.container);
+    this.element.appendChild(this.outerContainer);
+    document.body.appendChild(this.element);
+
+    this.element.addEventListener('opened', this);
+    this.element.addEventListener('closed', this);
+    this.smartBubble.addEventListener('all-items-bubbled', this);
+  }
+
+  var proto = {};
+
+  proto.open = function(message, buttonSettings, onCancel) {
+    this.isOpened = true;
+    message = message || '';
+
+    // onCancel is triggered when ESC key is pressed.
+    this.onCancel = onCancel;
+    this.buttonElements = [];
+    this.buttonSettings = buttonSettings;
+    this.messageElement.textContent = message;
+    this.messageElement.setAttribute('data-l10n-id', message);
+    this.buttonGroup.innerHTML = '';
+
+    this.messageElement.classList.remove('hidden');
+    if (message.length === 0) {
+      this.messageElement.classList.add('hidden');
+    }
+
+    // Set up every button
+    buttonSettings.forEach(function(buttonSetting, index) {
+      var button = document.createElement('smart-button');
+      button.setAttribute('type', 'circle-text');
+      button.setAttribute('data-l10n-id', buttonSetting.text || 'OK');
```

Comment:
We will use lower case in l10n id, like `ok` instead of `OK`.

It's the same here. Please use textRaw and textL10nId to denote the raw and id.